File: src/goto.c
#include "include/goto.h"
#include "include/cursor.h"
#include "include/editor.h"

void
goto__start_line() {
    cursor_t *cursor = editor__get_cursor();

    cursor__from(0, cursor->y);
}

void
goto__end_line() {
    cursor_t *cursor = editor__get_cursor();
    str_t *line = editor__get_current_line();

    if (line != NULL) {
        cursor__from(line->len, cursor->y);
    }
}

void
goto__start_file() {
    cursor__from(0, 0);
}

void
goto__end_file() {
    cursor__from(0, EDITOR.buffer.lines);
}

File: src/goto.go
package main

func GotoStartLine() {
	cursor := EDITOR.GetCursor()

	cursor.From(0, cursor.y)
}

func GotoEndLine() {
	cursor := EDITOR.GetCursor()
	line := EDITOR.GetCurrentLine()

	if line != nil {
		cursor.From(int64(len(line)), cursor.y)
	}
}

func GotoStartFile() {
	cursor := EDITOR.GetCursor()

	cursor.From(0, 0)
}

func GotoEndFile() {
	cursor := EDITOR.GetCursor()

	cursor.From(0, EDITOR.buffer.lines)
}
